src/Backend/SgateShopgatePlugin/Controllers/Frontend/shopgate.php

```diff
@@ -1006,6 +1281,654 @@ public function registrationAction()
+            }
+            $orderNumber = $cartItem->getProductOrderNumber();
+            $product = $this->prepareArticle($orderNumber, $cartItem);
+
+            if ($product) {
+                $items[] = $product;
+            }
+        }
+
+        return $items;
+    }
+
+    /**
+     * A helper function that returns a prepared product object ready to be displayed in the frontend
+     *
+     * @param int      $orderNumber
+     * @param CartItem $cartItem
+     *
+     * @return null|array
+     */
+    private function prepareArticle($orderNumber, $cartItem)
+    {
+        if (!$orderNumber) {
+            return null;
+        }
+
+        //Gets the product by the order number
+        $productId = $this->get('modules')->Articles()->sGetArticleIdByOrderNumber($orderNumber);
+
+        if (!$productId) {
+            return null;
+        }
+
+        try {
+            $product = $this->get('modules')->Articles()->sGetArticleById($productId, null, $orderNumber);
+        } catch (\RuntimeException $e) {
+            // if product is not found the ProductNumberService will throw an exception
+            return null;
+        }
+
+        if (!$product || !$this->existsInMainCategory($productId)) {
+            return null;
+        }
+
+        $sql = 'SELECT active FROM s_articles_details WHERE ordernumber = :orderNumber;';
+        $isVariantActive = $this->get('db')->fetchOne($sql, array('orderNumber' => $orderNumber));
+
+        if (!$isVariantActive) {
+            $message = $this->get('snippets')->getNamespace('frontend/plugins/swag_advanced_cart/plugin')->get('ArticleNotAvailable');
+            $items[] = array(
+                'id' => $cartItem->getId(),
+                'ordernumber' => $cartItem->getProductOrderNumber(),
+                'quantity' => $cartItem->getQuantity(),
+                'name' => $message,
+                'article' => array(
+                    'articleName' => $message,
+                    'articlename' => $message,
+                ),
+            );
+
+            return null;
+        }
+
+        $product['price'] = str_replace(',', '.', $product['price']) * $cartItem->getQuantity();
+
+        //For compatibility issues
+        $product['articlename'] = $product['articleName'];
+
+        if ($product['sConfigurator']) {
+            if ($product['additionaltext']) {
+                $product['articlename'] .= ' ' . $product['additionaltext'];
+                $product['articleName'] .= ' ' . $product['additionaltext'];
+            } else {
+                $productName = $this->get('modules')->Articles()->sGetArticleNameByOrderNumber($orderNumber);
+                $product['articlename'] = $productName;
+                $product['articleName'] = $productName;
+            }
+        }
+
+        $product['img'] = $product['image']['src'][0];
+
+        return $product;
+    }
+
+    /**
+     * check for product exists in active category
+     *
+     * @param $productId
```

Comment:
make sure to indicate all parameters, what should it be, is it an int or a string?